app/Helpers/database/forum.php

```diff
@@ -438,10 +439,18 @@ function authorizeAllForumPostsForUser(User $user): bool
 
 function isUserSubscribedToForumTopic(int $topicID, int $userID): bool
 {
-    return isUserSubscribedTo(
-        SubscriptionSubjectType::ForumTopic,
-        $topicID,
-        $userID,
-        "SELECT 1 FROM ForumTopicComment WHERE ForumTopicID = $topicID AND author_id = $userID"
-    );
+    $explicitSubcription = Subscription::where('subject_type', SubscriptionSubjectType::ForumTopic)
+        ->where('subject_id', $topicID)
+        ->where('user_id', $userID)
+        ->first();
+
+    if ($explicitSubcription) {
+        return $explicitSubcription->state == 1;
```

Comment:
In the `Subscription` model, I see:
```php
protected $casts = [
    'state' => 'boolean',
];
```

And when I `dd($explicitSubscription->toArray())`, I see:
```
array:7 [▼ // app/Helpers/database/forum.php:448
  "id" => 65828
  "subject_type" => "ForumTopic"
  "subject_id" => 25399
  "user_id" => 117089
  "state" => false
  "created_at" => "2024-09-15T12:00:44.000000Z"
  "updated_at" => "2024-09-15T12:00:44.000000Z"
]
```

I think it's probably safer to change this line to:
```php
return $explicitSubscription->state === true;
```

Reply:
Didn't think to see if the model coerced the type. Was just looking at the database. Updated.